internals/cli/cmd_notices.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/canonical/go-flags"
+
+	"github.com/canonical/pebble/client"
+	"github.com/canonical/pebble/internals/osutil"
+)
+
+const (
+	noticesFilenameEnvKey  = "PEBBLE_NOTICES_FILENAME"
+	noticesFilenameDefault = "~/.pebble/notices_<socket-path>.json"
+)
+
+const cmdNoticesSummary = "List notices"
+const cmdNoticesDescription = `
+The notices command lists the notices that have occurred, ordered by the
+Repeated time (oldest first).
+
+If --timeout is given and matching notices aren't yet available, wait up to
+the given duration for matching notices to arrive.
+
+By default, this lists all notices. If 'pebble ack' has been executed, this
+will only list the notices that have occurred (or been repeated) since the
+last one listed before the 'pebble ack'.
```

Comment:
Again I think we have to either correct my understanding :) or make this easier to understand. My opinion is that the ```repeated``` mechanism is not what it appears in text, and as a result what this states and what happens are different.

I will give my version:

```
By default, this lists all notices. If 'pebble ack' has been executed, this will record the most recently listed ('pebble notices') notice time, and as a result, future 'pebble notices' commands will only list announced notices after this point in time.
```

Ok this does not help, but what I am trying to say is that repeat to the reader will suggest a reoccurance of the notice. This is not true, because the ```repeatAfter``` attribute will effectively hide some of the notices, but only to a noticewaiter, which is this command!

Reply:
Thinking about this again, I think the complexity of this requires proper documentation with examples.

Reply:
See above about my README documentation, which explains "repeated" further. Let me know if that still doesn't help.